### PR TITLE
[DEVOPS-1220]  Introduces separation of taskboard project and jira project

### PR DESCRIPTION
**Pull request by taksan**

### CODE REVIEW

File: core/src/main/java/objective/taskboard/data/Issue.java [hunk 1]
```diff
@@ -384,10 +385,19 @@ public Long getId() {
...
         return this.projectKey;
     }
+    
+    public String getJiraProjectKey() {
+        return this.jiraProjectKey;
+    }
 
     public String getProject() {
+        if (this.projectKey == null) 
+            return getParentCard().map(p->p.getProject()).orElse(project);
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/data/Issue.java]: MINOR Replace this lambda with a method reference. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1612)

File: core/src/main/java/objective/taskboard/data/Issue.java [hunk 2]
```diff
@@ -384,10 +385,19 @@ public Long getId() {
     }
 
     public String getProjectKey() {
+        if (this.projectKey == null) 
+            return getParentCard().map(p->p.getProjectKey()).orElse(this.jiraProjectKey);
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/data/Issue.java]: MINOR Replace this lambda with a method reference. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1612)

File: core/src/main/java/objective/taskboard/domain/converter/JiraIssueToIssueConverter.java [hunk 3]
```diff
@@ -106,12 +107,30 @@ public IssueScratch createIssueScratch(JiraIssueDto jiraIssue) {
...
+        String taskbProjectKey = null;
+        String taskbProjectName = null;
+        if (taskboardProject.isPresent()) {
+            taskbProjectKey = taskboardProject.get().getProjectKey();
+            taskbProjectName = taskboardProject.get().getName();
+        } 
+        else {
+            Optional<ProjectFilterConfiguration> taskboardProjectWithSameKeyAsJiraProject = projectService.getTaskboardProject(jiraIssue.getProject().getKey());
+            if (projectService.getTaskboardProject(jiraIssue.getProject().getKey()).isPresent()) 
+                taskbProjectName = taskboardProjectWithSameKeyAsJiraProject.get().getName();
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/JiraIssueToIssueConverter.java]: MAJOR Call "taskboardProjectWithSameKeyAsJiraProject.isPresent()" before accessing the value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3655)

File: jira-client/src/main/java/objective/taskboard/jira/data/Version.java [hunk 4]
```diff
@@ -9,10 +11,13 @@
 
     public String id;
     public String name;
+    public boolean archived;
```
Comment (objective-solutions-user) [inline jira-client/src/main/java/objective/taskboard/jira/data/Version.java]: MINOR Make archived a static final constant or non-public and provide accessors if needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: jira-client/src/main/java/objective/taskboard/jira/data/Version.java [hunk 5]
```diff
@@ -9,10 +11,13 @@
...
+    public boolean released;
```
Comment (objective-solutions-user) [inline jira-client/src/main/java/objective/taskboard/jira/data/Version.java]: MINOR Make released a static final constant or non-public and provide accessors if needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: core/src/main/java/objective/taskboard/controller/ProjectCreationData.java [hunk 6]
```diff
@@ -27,6 +27,7 @@
     public String teamLeader;
     public Long baseClusterId;
     public ProjectCreationDataTeam defaultTeam;
+    public String projectName;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/controller/ProjectCreationData.java]: MINOR Make projectName a static final constant or non-public and provide accessors if needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: core/src/main/java/objective/taskboard/domain/converter/JiraIssueToIssueConverter.java [hunk 7]
```diff
@@ -106,12 +107,29 @@ public IssueScratch createIssueScratch(JiraIssueDto jiraIssue) {
...
+        
+        String taskbProjectKey = null;
+        String taskbProjectName = null;
+        if (taskboardProject.isPresent()) {
+            taskbProjectKey = taskboardProject.get().getProjectKey();
+            taskbProjectName = taskboardProject.get().getName();
+        } 
+        else {
+            Optional<ProjectFilterConfiguration> taskboardProjectWithSameKeyAsJiraProject = projectService.getTaskboardProject(jiraIssue.getProject().getKey());
+            taskbProjectName = taskboardProjectWithSameKeyAsJiraProject.map(p->p.getName()).orElse(jiraIssue.getProject().getName());
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/JiraIssueToIssueConverter.java]: MINOR Replace this lambda with a method reference. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1612)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: jira-client/src/main/java/objective/taskboard/jira/data/Version.java [hunk 8]
```diff
@@ -11,8 +11,8 @@
 
     public String id;
     public String name;
-    public boolean archived;
-    public boolean released;
+    public boolean archived;//NOSONAR
+    public boolean released;//NOSONAR
```
Comment (jhonypg) [inline jira-client/src/main/java/objective/taskboard/jira/data/Version.java]: O que o Sonar reclamou, nesse caso?

File: sizing-importer/src/main/java/objective/taskboard/sizingImport/SizingImportController.java [hunk 9]
```diff
@@ -134,7 +134,8 @@ public void postGoogleAuthorizarionCode(@RequestBody String authorizationCode) {
                 .collect(toList());
 
         try {
-            sizingImportService.importSpreadsheet(projectKey, spreadsheetId, dynamicColumnsMapping);
+            ProjectFilterConfiguration project = projectService.getTaskboardProjectOrCry(projectKey);
+            sizingImportService.importSpreadsheet(project.getJiraProjectKey(), spreadsheetId, dynamicColumnsMapping);
```
Comment (jhonypg) [inline sizing-importer/src/main/java/objective/taskboard/sizingImport/SizingImportController.java]: Muda no importSpreadsheet de projectKey para jiraProjectKey.

**Pull request merged**